docs/DeveloperGuide.md

```diff
@@ -2,37 +2,121 @@
+- Save each description name and nutrient/calories information in the corresponding hashmap (meal, drink, and exercise) to be used by other functions. 
+
+### Saved Meal, Drink, Exercises Storage
+- Each meal object in the meal list corresponds to a string with a format of `meal_name,serving_size,date`
+- Each drink object in the drink list corresponds to a string with a format of `drink_name,volume,date`
+- Each exercise object in the exercise list corresponds to a string with a format of `exercise_name,duration,intensity,date`
+- To store, convert all objects (meal, drink, exercise) in the list into its corresponding string, then write the appended strings into a .txt file
+- To retrieve, parse each string using "," as its delimiter and convert it into its corresponding objects, then add all the entries to the object list (mealList, drinkList, exerciseList)
 
 
 ## Product scope
 ### Target user profile
-
-{Describe the target user profile}
+- Have a need to manage their dietary intake and exercise routines effectively.
+- Prefer desktop applications over other types of platforms.
+- Can type quickly and prefer typing over mouse interactions.
+- Are reasonably comfortable using command-line interface (CLI) applications.
 
 ### Value proposition
 
-{Describe the value proposition: what problem does it solve?}
+The fitness app aims to help users manage their dietary habits and exercise routines more efficiently compared to traditional GUI-driven apps. 
+By offering a streamlined interface optimized for keyboard input and CLI interactions, users can track their meals, drinks, and exercises swiftly, allowing them to focus more on their fitness and nutritional goals and less on navigating through complex user interfaces.
 
 ## User Stories
 
-|Version| As a ... | I want to ... | So that I can ...|
-|--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+| Priority | As a ... | I want to ...          | So that I can ...                                      |
+|----------|----------|------------------------|--------------------------------------------------------|
+| ***      |new user| see usage instructions | refer to them when I forget how to use the application |
+| ***      |user| add a meal | track my daily calorie intake                          |
 
 ## Non-Functional Requirements
 
-{Give non-functional requirements}
+1. Should work on any mainstream OS (Linux, Windows, MacOS) as long as it has Java 11 or above installed.
+2. A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
 
```

Comment:
Give a summary of the commands required to be input by the user so that he/she can start using the app for typical usage